src/sentry/snuba/metrics/query.py

```diff
@@ -45,6 +45,16 @@ def __str__(self) -> str:
         return f"{self.op}({self.metric_name})" if self.op else self.metric_name
 
 
+@dataclass(frozen=True)
+class MetricGroupByField:
+    name: str
```

Comment:
so you decided to add`MetricGroupByField` and not go with Union[str|MetricField]?

Do we still need to take care of something like it?
```
MetricGroupByField(
                    MetricField(
                        op=None,
                        metric_name="e:transactions/team_key_transactions@none",
                        args=[[(20, "foo_transaction")]],
                        alias="e:transactions/team_key_transactions@none",
                    ),
                ),
```

Reply:
Let's address that in a subsequent PR. No need to complicate this one any further

Reply:
Sure